src/api/settings/getServerSettings.js

```diff
@@ -52,6 +53,47 @@ export type ApiResponseServerSettings = {|
   realm_web_public_access_enabled: boolean,
 |};
 
+type ServerApiResponseServerSettings = {|
+  ...ApiResponseServerSettings,
+
+  // When missing, the user requested the root domain of a Zulip server, but
+  // there is no realm there. User error.
+  //
+  // (Also, the server has `ROOT_DOMAIN_LANDING_PAGE = False`, the default.
+  // But the mobile client doesn't care about that; we just care that there
+  // isn't a realm at the root.)
+  //
+  // Discussion:
+  //   https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/.2Fserver_settings.3A.20.60realm_name.60.2C.20etc.2E/near/1334042
+  //
+  // TODO(server-future): Expect the error to be encoded in a proper error
+  //   response instead of this missing property.
+  // TODO(server-future): Then, when all supported servers give that error,
+  //   make this property required.
+  realm_name?: string,
+|};
+
 /** See https://zulip.com/api/get-server-settings */
-export default async (realm: URL): Promise<ApiResponseServerSettings> =>
-  apiGet({ apiKey: '', email: '', realm }, 'server_settings');
+export default async (realm: URL): Promise<ApiResponseServerSettings> => {
+  const result: ServerApiResponseServerSettings = await apiGet(
+    { apiKey: '', email: '', realm },
+    'server_settings',
+  );
+
+  const { realm_name } = result;
+  if (realm_name == null) {
+    // See comment on realm_name.
```

Comment:
nit: this sounds like it means the local variable of that name; "on realm_name in ServerApiResponseServerSettings" would make it clear.